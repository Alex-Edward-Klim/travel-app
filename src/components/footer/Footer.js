import React from "react";
import "./footer.scss" 
import logoSrc from "../../images/rs_school_js.svg";

function Footer() {
    return(
        <footer className="footer">
            <div className="footer__container">
                <a className="footer__logo" href="https://rs.school/js/">
                    <img className="footer__logo__img" alt="rs school" src={logoSrc} />
                </a>
                <p className="footer__authors">
                    <span>by</span>
                    <a href="https://github.com/Alex-Edward-Klim"> Aleksey Klimovskoy</a>,
                    <a href="https://github.com/korytsa"> Anna Korytko</a>,
                    <a href="https://github.com/YaroslavTrefilov"> Yaroslav Trefilov</a>,
                    <a href="https://github.com/NachinkaShaurmi"> Iurii Oleinik</a>
                </p>
                <a className="footer__rss" href="https://rs.school/js/">
                    <p>RS School</p>
                    <p>2021</p>
                </a>
            </div>
        </footer>
    )
}

export default Footer;