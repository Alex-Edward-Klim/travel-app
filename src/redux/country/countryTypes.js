// Action Types
export const ADD_COUNTRY = "ADD_COUNTRY";
