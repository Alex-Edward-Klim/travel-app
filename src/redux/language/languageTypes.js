// Action Types
export const CHANGE_LANGUAGE_TO_EN = "CHANGE_LANGUAGE_TO_EN";
export const CHANGE_LANGUAGE_TO_RU = "CHANGE_LANGUAGE_TO_RU";
export const CHANGE_LANGUAGE_TO_BE = "CHANGE_LANGUAGE_TO_BE";
