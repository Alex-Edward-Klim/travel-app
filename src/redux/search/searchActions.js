import {
  CHANGE_SEARCH_TO_EMPTY,
  CHANGE_SEARCH_TO_VALUE,
} from "./searchTypes";

// Action Creators

export const changeSearchToEmpty = () => ({
  type: CHANGE_SEARCH_TO_EMPTY,
});

export const changeSearchToValue = (e) => ({
  type: CHANGE_SEARCH_TO_VALUE,
  payload: e.target.value
});



export const changeSearch = (action, e = '') => {
  switch (action) {
    case "clear":
      return changeSearchToEmpty();
    case "set":
      return changeSearchToValue(e);
  }
};
