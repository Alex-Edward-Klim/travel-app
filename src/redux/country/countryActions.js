import { ADD_COUNTRY } from "./countryTypes";

// Action Creators

export const addCountry = (country) => ({
  type: ADD_COUNTRY,
  payload: country
});
