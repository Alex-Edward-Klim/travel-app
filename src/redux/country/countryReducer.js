import { FETCH_COUNTRIES_FAILURE, FETCH_COUNTRIES_REQUEST, FETCH_COUNTRIES_SUCCESS } from "./countryTypes";

// Initial State
export const initialState = {
  loading: false,
  countries: [],
  error: ''
};

// Reducer
export const countryReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_COUNTRIES_REQUEST:
      return {
        ...state,
        loading: true
      };
    case FETCH_COUNTRIES_SUCCESS:
    return {
      ...state,
      loading: false,
      countries: action.payload,
      error: ''
    };
    case FETCH_COUNTRIES_FAILURE:
    return {
      ...state,
      loading: false,
      countries: [],
      error: action.payload
    };
    default:
      return state;
  };
};

export default countryReducer;
