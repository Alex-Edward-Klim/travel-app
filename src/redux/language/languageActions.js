import { CHANGE_LANGUAGE_TO_BE, CHANGE_LANGUAGE_TO_EN, CHANGE_LANGUAGE_TO_RU } from "./languageTypes";

// Action Creators

export const changeLanguageToEn = () => ({
  type: CHANGE_LANGUAGE_TO_EN,
});

export const changeLanguageToRu = () => ({
  type: CHANGE_LANGUAGE_TO_RU,
});

export const changeLanguageToBe = () => ({
  type: CHANGE_LANGUAGE_TO_BE,
});

export const changeLanguage = (lang) => {
  switch (lang) {
    case "EN":
      return changeLanguageToEn();
    case "RU":
    return changeLanguageToRu();
    case "BE":
    return changeLanguageToBe();
    // default:
    //   return;
  };

};
