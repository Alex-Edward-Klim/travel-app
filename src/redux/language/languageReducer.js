import { CHANGE_LANGUAGE_TO_BE, CHANGE_LANGUAGE_TO_EN, CHANGE_LANGUAGE_TO_RU } from "./languageTypes";

// Initial State
export const initialState = {
  language: "EN",
};

const storageLang = localStorage.getItem("TravelAppUserLanguageData78fe8a83ef752bd23c98c262b7264947");

if (storageLang) {
  initialState.language = storageLang;
} else {
  localStorage.setItem("TravelAppUserLanguageData78fe8a83ef752bd23c98c262b7264947", "EN");
}

// Reducer
export const languageReducer = (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_LANGUAGE_TO_EN:
      return {
        ...state,
        language: "EN"
      };
    case CHANGE_LANGUAGE_TO_RU:
    return {
      ...state,
      language: "RU"
    };
    case CHANGE_LANGUAGE_TO_BE:
    return {
      ...state,
      language: "BE"
    };
    default:
      return state;
  };
};

export default languageReducer;
