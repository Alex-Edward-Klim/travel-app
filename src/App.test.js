import * as language from "./redux/language/languageReducer";
import * as languageType from "./redux/language/languageTypes";
import * as search from "./redux/search/searchReducer";
import * as searchType from "./redux/search/searchTypes";
import * as country from "./redux/country/countryReducer";
import * as countryType from "./redux/country/countryTypes";
import { render } from "@testing-library/react";
import Footer from "./components/footer/Footer";

describe("initial states", () => {
  it("language initial", () => {
    expect(language.initialState.language).toBe("EN");
  });
  it("country initial", () => {
    expect(country.initialState.loading).toBe(false);
    expect(country.initialState.error).toBe('');
    expect(country.initialState.countries).toHaveLength(0);
  });
  it("search initial", () => {
    expect(search.initialState.search).toBe("");
  });

});

describe("language reducer", () => {
  it("CHANGE_LANGUAGE_TO_RU", () => {
    const action = {
      type: languageType.CHANGE_LANGUAGE_TO_RU,
    };

    expect(language.languageReducer(language.initialState, action)).toEqual({
      language: "RU",
    });
  });

  it("CHANGE_LANGUAGE_TO_EN", () => {
    const action = {
      type: languageType.CHANGE_LANGUAGE_TO_EN,
    };

    expect(language.languageReducer(language.initialState, action)).toEqual({
      language: "EN",
    });
  });

  it("CHANGE_LANGUAGE_TO_BE", () => {
    const action = {
      type: languageType.CHANGE_LANGUAGE_TO_BE,
    };

    expect(language.languageReducer(language.initialState, action)).toEqual({
      language: "BE",
    });
  });
});

describe("search reducer", () => {
  it("CHANGE_SEARCH_TO_EMPTY", () => {
    const action = {
      type: searchType.CHANGE_SEARCH_TO_EMPTY,
    };

    expect(search.searchReducer("England", action)).toMatchObject({
      search: "",
    });
  });

  it("CHANGE_SEARCH_TO_VALUE", () => {
    const action = {
      type: searchType.CHANGE_SEARCH_TO_VALUE,
      payload: "testString",
    };

    expect(search.searchReducer(search.initialState, action)).toMatchObject({
      search: "testString",
    });
  });
});

describe("country reducer", () => {
  it("FETCH_COUNTRIES_REQUEST", () => {
    const action = {
      type: countryType.FETCH_COUNTRIES_REQUEST,
    };

    expect(country.countryReducer(country.initialState, action)).toMatchObject({
      loading: true,
    });
  });

  it("FETCH_COUNTRIES_SUCCESS", () => {
    const action = {
      type: countryType.FETCH_COUNTRIES_SUCCESS,
    };
    expect(
      country.countryReducer(
        {
          loading: true,
          countries: [],
          error: "test",
        },
        action
      )
    ).toMatchObject({
      loading: false,
      error: "",
    });
  });

  it("FETCH_COUNTRIES_FAILURE", () => {
    const action = {
      type: countryType.FETCH_COUNTRIES_FAILURE,
    };
    expect(
      country.countryReducer(
        {
          loading: true,
          countries: ['a', 'b', 'c'],
          error: "test",
        },
        action
      )
    ).toMatchObject({
      loading: false,
      countries: [],
    });
  });
});

describe("Snapshot component", () => {
  it("Footer", () => {
    const rendered = render(<Footer />);
    expect(rendered).toMatchSnapshot();
  });
});
