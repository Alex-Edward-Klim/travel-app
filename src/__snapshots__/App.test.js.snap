// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot component Footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <div
          class="footer__container"
        >
          <a
            class="footer__logo"
            href="https://rs.school/js/"
          >
            <img
              alt="rs school"
              class="footer__logo__img"
              src="rs_school_js.svg"
            />
          </a>
          <p
            class="footer__authors"
          >
            <span>
              by
            </span>
            <a
              href="https://github.com/Alex-Edward-Klim"
            >
               Aleksey Klimovskoy
            </a>
            ,
            <a
              href="https://github.com/korytsa"
            >
               Anna Korytko
            </a>
            ,
            <a
              href="https://github.com/YaroslavTrefilov"
            >
               Yaroslav Trefilov
            </a>
            ,
            <a
              href="https://github.com/NachinkaShaurmi"
            >
               Iurii Oleinik
            </a>
          </p>
          <a
            class="footer__rss"
            href="https://rs.school/js/"
          >
            <p>
              RS School
            </p>
            <p>
              2021
            </p>
          </a>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <div
        class="footer__container"
      >
        <a
          class="footer__logo"
          href="https://rs.school/js/"
        >
          <img
            alt="rs school"
            class="footer__logo__img"
            src="rs_school_js.svg"
          />
        </a>
        <p
          class="footer__authors"
        >
          <span>
            by
          </span>
          <a
            href="https://github.com/Alex-Edward-Klim"
          >
             Aleksey Klimovskoy
          </a>
          ,
          <a
            href="https://github.com/korytsa"
          >
             Anna Korytko
          </a>
          ,
          <a
            href="https://github.com/YaroslavTrefilov"
          >
             Yaroslav Trefilov
          </a>
          ,
          <a
            href="https://github.com/NachinkaShaurmi"
          >
             Iurii Oleinik
          </a>
        </p>
        <a
          class="footer__rss"
          href="https://rs.school/js/"
        >
          <p>
            RS School
          </p>
          <p>
            2021
          </p>
        </a>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
